kubernetes/data_source_kubernetes_server_version.go

```diff
@@ -0,0 +1,40 @@
+package kubernetes
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gversion "github.com/hashicorp/go-version"
+)
+
+func dataSourceKubernetesServerVersion() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceKubernetesServerVersionRead,
+		Schema: map[string]*schema.Schema{
+			"version": {
+				Type:        schema.TypeString,
+				Description: "Kubernetes cluster version.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceKubernetesServerVersionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn, err := meta.(KubeClientsets).MainClientset()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	sv, err := conn.ServerVersion()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	gv, err := gversion.NewVersion(sv.String())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(gv.String())
+	d.Set("version", gv.String())
```

Comment:
Was gonna open a PR and saw yours. Mine had extra version information to match the `kubectl version` command that can be added here.

```
	server_version := map[string]string{
		"buildDate":    sv.BuildDate,
		"compiler":     sv.Compiler,
		"gitCommit":    sv.GitCommit,
		"gitTreeState": sv.GitTreeState,
		"gitVersion":   sv.GitVersion,
		"goVersion":    sv.GoVersion,
		"major":        sv.Major,
		"minor":        sv.Minor,
		"platform":     sv.Platform,
		"version":      sv.String(),
	}
	d.Set("server_version", server_version)
```

Reply:
This is a good one. I thought about this as well. Let me add this to my PR if you don't mind.